### DISCUSSION TITLE
Cannot connect to local database using NextJS

### DISCUSSION THREAD

**Discussion by blacksandsolutions** [supabase/supabase#19578] [Questions] [answered]
I've been working to date with the hosted project, but attempted to switch to a local env today.
I have the local DB linked and migrated and have updated my ENV vars

```
NEXT_PUBLIC_SUPABASE_URL=postgresql://postgres:postgres@127.0.0.1:54322/postgres
```
However when I try to read from the DB I see this error in the logs

`Request cannot be constructed from a URL that includes credentials`

![Screenshot 2023-12-11 at 1 28 57 PM](https://github.com/supabase/supabase/assets/58924459/08f39ebd-8ac9-44f2-ba6c-2b23e467e7df)

The request is made from a NextJS Server component, however I think the error is Supabase related - the client maybe?

**Comment by GaryAustin1** [comment, marked as answer]
You are using the direct database port url in a supabase-js client I assume.  That is not the correct URL.

![image](https://github.com/supabase/supabase/assets/54564956/9a5f0f07-75c6-4fd9-95b7-74193502adb6)

**Comment by blacksandsolutions** [comment]
@GaryAustin1 Thanks for the quick response.

OK, I update the ENV and now have
```
NEXT_PUBLIC_SUPABASE_URL=http://127.0.0.1:54321
NEXT_PUBLIC_SUPABASE_ANON_KEY=xxxxxxxxx
```

But now get a new error
![Screenshot 2023-12-11 at 2 22 35 PM](https://github.com/supabase/supabase/assets/58924459/9efe534e-01e5-4e73-9e86-75aae5b556db)

Which I guess is because I have not set JWT secret in the ENV?

But then again, the code to create client does not accept a JWT, unless it goes in options?
```ts
   createBrowserClient(
        process.env.NEXT_PUBLIC_SUPABASE_URL!,
        process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY!,
    )
```

Is there any documentation you could link to for this?

**Reply by GaryAustin1** [reply]
No idea as you are not showing any code.  JWT secret is not involved in the clients.   Could be your anon key is not correct.  You should log it out or check it with jwt.io.

After that all bets are off as at some point I assume you are making a database call, but have no idea if you have attempted to sign in a user and if that session is in the client.  

So either your anon key is bad or your user session access_token is bad is what the error says to me.

**Reply by blacksandsolutions** [reply]
Doh! Yes Anon Key was wrong. I pasted the new one for local into the wrong ENV file.

That's fixed it, thanks. 

My database has no data in it though which surprised me. So I guess the linking and migrating only updates the schema - Ill have to read through those docs again.

**Reply by GaryAustin1** [reply]
Please ask a new question if you have issues with that.  I personally don't use local instances.
